From 50f531462274bb6c395ff1b63cf7c1016d7bb088 Mon Sep 17 00:00:00 2001
From: David Robillard <d@drobilla.net>
Date: Thu, 11 Mar 2021 13:55:08 -0500
Subject: [PATCH] Remove unused datatypes from xsd.ttl

This data costs memory, and I think it is better to not present things that
aren't useful to LV2 implementations at all here.

The whitespace annotations could be useful for cleaning up literals, but this
isn't currently used (and tends to be baked in to implementations anyway), so
remove them as well.  If necessary they can be added later easily enough.
---
 schemas.lv2/xsd.ttl | 152 +++++---------------------------------------
 1 file changed, 16 insertions(+), 136 deletions(-)

diff --git a/schemas.lv2/xsd.ttl b/schemas.lv2/xsd.ttl
index 47030624..007d84ed 100644
--- a/schemas.lv2/xsd.ttl
+++ b/schemas.lv2/xsd.ttl
@@ -7,52 +7,15 @@ xsd:
 	a owl:Ontology ;
 	rdfs:comment "XML Schema Datatypes" .
 
-xsd:ENTITY
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:NCName .
-
-xsd:ID
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:NCName .
-
-xsd:IDREF
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:NCName .
-
-xsd:NCName
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:Name .
-
-xsd:NMTOKEN
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:token .
-
-xsd:Name
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:token .
-
-xsd:QName
-	a rdfs:Datatype ;
-	rdfs:label "XML qualified name" ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
-
 xsd:anySimpleType
-	a rdfs:Datatype .
+	a rdfs:Datatype ;
+	rdfs:comment "The base class of any primitive XSD dataype." ;
+	rdfs:label "any simple type" .
 
 xsd:anyURI
 	a rdfs:Datatype ;
-	rdfs:label "URI reference" ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
+	rdfs:label "any URI" ;
+	owl:onDatatype xsd:anySimpleType .
 
 xsd:base64Binary
 	a rdfs:Datatype ;
@@ -62,8 +25,6 @@ xsd:base64Binary
 	owl:withRestrictions (
 		[
 			xsd:pattern "(([A-Za-z0-9+/] *[A-Za-z0-9+/] *[A-Za-z0-9+/] *[A-Za-z0-9+/] *)*(([A-Za-z0-9+/] *[A-Za-z0-9+/] *[A-Za-z0-9+/] *[A-Za-z0-9+/])|([A-Za-z0-9+/] *[A-Za-z0-9+/] *[AEIMQUYcgkosw048] *=)|([A-Za-z0-9+/] *[AQgw] *= *=)))?"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -74,8 +35,6 @@ xsd:boolean
 	owl:withRestrictions (
 		[
 			xsd:pattern "(true|false|0|1)"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -85,9 +44,9 @@ xsd:byte
 	owl:onDatatype xsd:short ;
 	owl:withRestrictions (
 		[
-			xsd:minInclusive -128
-		] [
 			xsd:maxInclusive 127
+		] [
+			xsd:minInclusive -128
 		]
 	) .
 
@@ -98,8 +57,6 @@ xsd:date
 	owl:withRestrictions (
 		[
 			xsd:pattern "-?[0-9][0-9][0-9][0-9][0-9]*-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])(Z|[-+][0-2][0-9]:[0-5][0-9])?"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -110,8 +67,6 @@ xsd:dateTime
 	owl:withRestrictions (
 		[
 			xsd:pattern "-?[0-9][0-9][0-9][0-9][0-9]*-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])T(([0-1][0-9])|(2[0-4])):[0-5][0-9]:[0-5][0-9](.[0-9]+)?(Z|[-+][0-2][0-9]:[0-5][0-9])?"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -123,8 +78,6 @@ xsd:decimal
 	owl:withRestrictions (
 		[
 			xsd:pattern "-?INF|NaN|[+-]?(([0-9]+[.]?[0-9]*)|([0-9]*[.]?[0-9]+))([eE][-+]?[0-9]+)?"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -136,8 +89,6 @@ xsd:double
 	owl:withRestrictions (
 		[
 			xsd:pattern "-?INF|NaN|[+-]?(([0-9]+[.]?[0-9]*)|([0-9]*[.]?[0-9]+))([eE][-+]?[0-9]+)?"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -173,51 +124,6 @@ xsd:fractionDigits
 	rdfs:label "fraction digits" ;
 	rdfs:range xsd:nonNegativeInteger .
 
-xsd:gDay
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
-
-xsd:gMonth
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
-
-xsd:gMonthDay
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
-
-xsd:gYear
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
-
-xsd:gYearMonth
-	a rdfs:Datatype ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
-
 xsd:hexBinary
 	a rdfs:Datatype ;
 	rdfs:comment "Hex-encoded arbitrary binary data." ;
@@ -226,8 +132,6 @@ xsd:hexBinary
 	owl:withRestrictions (
 		[
 			xsd:pattern "([0-9A-Fa-f][0-9A-Fa-f])*"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -237,9 +141,9 @@ xsd:int
 	owl:onDatatype xsd:long ;
 	owl:withRestrictions (
 		[
-			xsd:minInclusive -2147483648
-		] [
 			xsd:maxInclusive 2147483647
+		] [
+			xsd:minInclusive -2147483648
 		]
 	) .
 
@@ -271,9 +175,9 @@ xsd:long
 	owl:onDatatype xsd:integer ;
 	owl:withRestrictions (
 		[
-			xsd:minInclusive -9223372036854775808
-		] [
 			xsd:maxInclusive 9223372036854775807
+		] [
+			xsd:minInclusive -9223372036854775808
 		]
 	) .
 
@@ -339,12 +243,7 @@ xsd:normalizedString
 	a rdfs:Datatype ;
 	rdfs:comment "The set of strings that do not contain the carriage return (#xD), line feed (#xA) nor tab (#x9) characters." ;
 	rdfs:label "normalized string" ;
-	owl:onDatatype xsd:string ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "replace"
-		]
-	) .
+	owl:onDatatype xsd:string .
 
 xsd:pattern
 	a rdf:Property ,
@@ -370,9 +269,9 @@ xsd:short
 	owl:onDatatype xsd:int ;
 	owl:withRestrictions (
 		[
-			xsd:minInclusive -32768
-		] [
 			xsd:maxInclusive 32767
+		] [
+			xsd:minInclusive -32768
 		]
 	) .
 
@@ -380,12 +279,7 @@ xsd:string
 	a rdfs:Datatype ;
 	rdfs:comment "A character string." ;
 	rdfs:label "string" ;
-	owl:onDatatype xsd:anySimpleType ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "preserve"
-		]
-	) .
+	owl:onDatatype xsd:anySimpleType .
 
 xsd:time
 	a rdfs:Datatype ;
@@ -394,8 +288,6 @@ xsd:time
 	owl:withRestrictions (
 		[
 			xsd:pattern "(([0-1][0-9])|(2[0-4])):[0-5][0-9]:[0-5][0-9](.[0-9]+)?(Z|[-+][0-2][0-9]:[0-5][0-9])?"
-		] [
-			xsd:whiteSpace "collapse"
 		]
 	) .
 
@@ -403,19 +295,7 @@ xsd:token
 	a rdfs:Datatype ;
 	rdfs:comment "The set of strings that do not contain the carriage return (#xD), line feed (#xA) nor tab (#x9) characters, that have no leading or trailing spaces (#x20) and that have no internal sequences of two or more spaces." ;
 	rdfs:label "token" ;
-	owl:onDatatype xsd:normalizedString ;
-	owl:withRestrictions (
-		[
-			xsd:whiteSpace "collapse"
-		]
-	) .
-
-xsd:totalDigits
-	a rdf:Property ,
-		owl:DatatypeProperty ;
-	rdfs:comment "The maximum number of decimal digits required to represent a value." ;
-	rdfs:label "total digits" ;
-	rdfs:range xsd:positiveInteger .
+	owl:onDatatype xsd:normalizedString .
 
 xsd:unsignedByte
 	a rdfs:Datatype ;
